### PR TITLE
Expose JoinedRoomsRepository and JoinedRoomsViewModel as protocols

### PR DESCRIPTION
**Pull request by grzesiekko**
### What?

This PR exposes `JoinedRoomsRepository` and `JoinedRoomsViewModel` in the public API as protocols.

### Why?

Increased testability.

----

### CODE REVIEW

File: Chatkit/Repositories/JoinedRoomsRepository.swift [hunk 1]
```diff
@@ -67,7 +93,7 @@ public class JoinedRoomsRepository: JoinedRoomsRepositoryProtocol {
         self.versionedState = buffer.currentState
         self.connectionState = connectivityMonitor.connectionState
         
-        self.state = JoinedRoomsRepository.state(forConnectionState: connectivityMonitor.connectionState,
+        self.state = Self.state(forConnectionState: connectivityMonitor.connectionState,
                                                  versionedState: buffer.currentState,
                                                  previousVersionedState: nil,
                                                  usingTransformer: dependencies.transformer)
```
Comment (OliverPearmain-Pusher) [inline Chatkit/Repositories/JoinedRoomsRepository.swift]: Really no big deal but there's a slight whitespace/padding mis-alignment as a result of your change.

File: Chatkit/Repositories/States/JoinedRoomsRepositoryState.swift [hunk 2]
```diff
@@ -1,10 +1,8 @@
 import Foundation
 
-public extension JoinedRoomsRepository {
-    
     /// An enumeration representing the state of a `JoinedRoomsRepository` which serves live data
     /// retrieved from the Chatkit web service.
-    enum State {
+    public enum JoinedRoomsRepositoryState {
```
Comment (OliverPearmain-Pusher) [inline Chatkit/Repositories/States/JoinedRoomsRepositoryState.swift]: Looks like this whole class might be pushed out by an extra tab stop.

File: Chatkit/Repositories/JoinedRoomsRepository.swift (hunk 1 above)
Comment (grzesiekko) [reply Chatkit/Repositories/JoinedRoomsRepository.swift]: I will fix that.

File: Chatkit/Repositories/JoinedRoomsRepository.swift [hunk 3]
```diff
@@ -105,10 +131,10 @@ public class JoinedRoomsRepository: JoinedRoomsRepositoryProtocol {
...
 
-extension JoinedRoomsRepository: BufferDelegate {
+extension ConcreteJoinedRoomsRepository: BufferDelegate {
     
     func buffer(_ buffer: Buffer, didUpdateState state: VersionedState) {
-        self.state = JoinedRoomsRepository.state(forConnectionState: self.connectionState,
+        self.state = Self.state(forConnectionState: self.connectionState,
                                                  versionedState: state,
                                                  previousVersionedState: self.versionedState,
                                                  usingTransformer: self.dependencies.transformer)
```
Comment (OliverPearmain-Pusher) [inline Chatkit/Repositories/JoinedRoomsRepository.swift]: Some more whitespace anomalies.

File: Chatkit/Repositories/JoinedRoomsRepository.swift [hunk 4]
```diff
@@ -119,38 +145,14 @@ extension JoinedRoomsRepository: BufferDelegate {
...
 
-extension JoinedRoomsRepository: ConnectivityMonitorDelegate {
+extension ConcreteJoinedRoomsRepository: ConnectivityMonitorDelegate {
     
     func connectivityMonitor(_ connectivityMonitor: ConnectivityMonitor, didUpdateConnectionState connectionState: ConnectionState) {
-        self.state = JoinedRoomsRepository.state(forConnectionState: connectionState,
+        self.state = Self.state(forConnectionState: connectionState,
                                                  versionedState: self.versionedState,
                                                  previousVersionedState: self.versionedState,
                                                  usingTransformer: self.dependencies.transformer)
```
Comment (OliverPearmain-Pusher) [inline Chatkit/Repositories/JoinedRoomsRepository.swift]: And here.

File: Chatkit/Repositories/States/JoinedRoomsRepositoryState.swift (hunk 2 above)
Comment (grzesiekko) [reply Chatkit/Repositories/States/JoinedRoomsRepositoryState.swift]: I will fix that.

File: Chatkit/Repositories/JoinedRoomsRepository.swift (hunk 3 above)
Comment (grzesiekko) [reply Chatkit/Repositories/JoinedRoomsRepository.swift]: I will fix that.

File: Chatkit/Repositories/JoinedRoomsRepository.swift (hunk 4 above)
Comment (grzesiekko) [reply Chatkit/Repositories/JoinedRoomsRepository.swift]: Fixed

File: Test Utilities/Doubles/Repositories/JoinedRoomsRepository.swift [hunk 5]
```diff
@@ -29,32 +29,13 @@ public class StubJoinedRoomsRepository: DoubleBase, JoinedRoomsRepositoryProtoco
...
-                                                              delegate_expectedSetCallCount: 1)
-        let stubTransformer = StubTransformer(room_toReturn: room, transformCurrentStatePreviousState_expectedSetCallCount: 1)
-        
-        let dependencies = DependenciesDoubles(transformer: stubTransformer)
-        
-        let joinedRoomsRepository = JoinedRoomsRepository(buffer: stubBuffer, connectivityMonitor: stubConnectivityMonitor, dependencies: dependencies)
-        
-        self.delegate?.joinedRoomsRepository(joinedRoomsRepository, didUpdateState: state)
+    public func report(_ state: JoinedRoomsRepositoryState) {
+        self.delegate?.joinedRoomsRepository(self, didUpdateState: state)
```
Comment (OliverPearmain-Pusher) [inline Test Utilities/Doubles/Repositories/JoinedRoomsRepository.swift]: Yay, much cleaner :-)

**Pull request merged**